tests/business/entities/rules/test_depth_average_rule.py

```diff
@@ -96,25 +96,27 @@ def test_depth_average_rule(
     logger = Mock(ILogger)
     rule = DepthAverageRule(
         name="test",
-        input_variable_names=["foo", "mesh2d_interface_z"],
-        layer_type="z",
+        input_variable_names=["foo",
+                              "mesh2d_flowelem_bl",
+                              "mesh2d_s1",
+                              "mesh2d_interface_z"],
     )
 
```

Comment:
Instead of fixed names, netCDF4/UGRID convention allows the retreival these names from the netCDF4 file itself. That would be much more robusted.

Reply:
the names are not fixed, in this test I just happened to use the same ones as FM uses. The user needs to supply which variable is used for the bottom, water level and interface.
When trying to retrieve those from the netCDF4 you would need to do it based on the attributes, which is really not so straightforward. Especially since Ugrid variables do not know for example if it is a z or sigma layer model.
we decided to go for the more convenient (read less code) approach, where the user defines which variable to use. especially since a user needs to know his/her model anyway